lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb

```diff
@@ -68,12 +68,26 @@ def generate
           # that use Swift. Setting this for the embedded target would
           # cause an App Store rejection because frameworks cannot be embedded
           # in embedded targets.
-          if !target.requires_host_target? && pod_targets.any?(&:uses_swift?)
-            config['EMBEDDED_CONTENT_CONTAINS_SWIFT'] = 'YES'
-            config['ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES'] = 'YES'
+
+	  # Extract swift_version number
+	  @swift_version = target.build_configurations.build_settings['SWIFT_VERSION']
+          @swift_version_check = 3.0
+
+	  # Check if swift_version is prior to v3.0
+	  if !target.requires_host_target? && pod_targets.any?(&:uses_swift?)
+	    if @swift_version < @swift_version_check
```

Comment:
nit: instead of doing this to decide the build setting in each branch, maybe pull this out above into something like:
```
if @swift_version < @swift_version_check
  embedded_swift_key = 'EMBEDDED_CONTENT_CONTAINS_SWIFT'
else
  embedded_swift_key = 'ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES'
end
```
then below you can just set `config[embedded_swift_key] = `…